### PR TITLE
islandora_audio plugin module

### PR DESCRIPTION
**Pull request by Natkeeran**
**GitHub Issue**: (link)
* https://github.com/Islandora-CLAW/CLAW/issues/930

# What does this Pull Request do?
Similar to video derivative plugin, this PR adds an audio derivative plugin.  

# What's new?
* islandora_audio sub module

# How should this be tested?
* Get this PR
* Go to modules and install the sub module: `/admin/modules`
* Go to actions: `/admin/config/system/actions`.  you should see a `Generate a audio derivative` option

# Additional Notes:
Would be good if you can review and see if video and audio can be abstracted into a single module.  It seems easier to keep them separate.

# Interested parties
@whikloj @MarcusBarnes  @dannylamb 
@Islandora-CLAW/committers

### CODE REVIEW

File: modules/islandora_audio/tests/src/Functional/GenerateAudioDerivativeTest.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      'bypass node access',
+      'administer contexts',
+      'administer actions',
+      'view media',
+      'create media',
+      'update media',
+    ]);
+    $this->drupalLogin($account);
+
+    // Create an action to generate a jpeg thumbnail.
```
Comment (dannylamb) [inline modules/islandora_audio/tests/src/Functional/GenerateAudioDerivativeTest.php]: Minor copy/pasta

File: modules/islandora_audio/src/Plugin/Action/GenerateAudioDerivative.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    $config = parent::defaultConfiguration();
+    $config['path'] = '[date:custom:Y]-[date:custom:m]/[node:nid].mp3';
+    $config['mimetype'] = 'audio/mpeg';
+    $config['queue'] = 'islandora-connector-homarus';
+    $config['destination_media_type'] = 'audio';
```
Comment (dannylamb) [inline modules/islandora_audio/src/Plugin/Action/GenerateAudioDerivative.php]: I'd go ahead and add the default args for making an mp3 wtih libmp3lame if we roll with keeping audio and video separate: `-codec:a libmp3lame -q:a 5`

**Comment by Natkeeran** [comment]
@dannylamb added the requested changes.  All code is by Jared.  Reusing it here.   Not even sure how those tests get executed.

**Comment by dannylamb** [comment]
It's all good man.  Those tests are a little crufty, but that's not on you (or @whikloj).  I'll clean them up a bit on a downcycle.

**Pull request merged**